src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -545,6 +551,70 @@ protected DynamoModel(IStartConfiguration config)
             LogWarningMessageEvents.LogWarningMessage += LogWarningMessage;
 
             StartBackupFilesTimer();
+
+            TraceReconciliationProcessor = this;
+        }
+
+        private void EngineController_TraceReconcliationComplete(TraceReconciliationEventArgs obj)
+        {
+            Debug.WriteLine("TRACE RECONCILIATION: {0} total serializables were orphaned.", obj.CallsiteToOrphanMap.SelectMany(kvp=>kvp.Value).Count());
+            
+            // The orphans will come back here as a dictionary of lists of ISerializables jeyed by their callsite id.
+            // This dictionary gets redistributed into a dictionary keyed by the workspace id.
+
+            var workspaceOrphanMap = new Dictionary<Guid, List<ISerializable>>();
+
+            foreach (var ws in Workspaces.OfType<HomeWorkspaceModel>())
+            {
+                // Get the orphaned serializables to this workspace
+                var wsOrphans = ws.GetOrphanedSerializablesAndClearHistoricalTraceData().ToList();
+
+                if (!wsOrphans.Any())
+                    continue;
+
+                if (!workspaceOrphanMap.ContainsKey(ws.Guid))
+                {
+                    workspaceOrphanMap.Add(ws.Guid, wsOrphans);
+                }
+                else
+                {
+                    workspaceOrphanMap[ws.Guid].AddRange(wsOrphans);
+                }
+            }
+
+            foreach (var kvp in obj.CallsiteToOrphanMap)
+            {
+                if (!kvp.Value.Any()) continue;
+
+                var nodeGuid = EngineController.LiveRunnerCore.DSExecutable.CallSiteToNodeMap[kvp.Key];
+
+                // Find the owning workspace for a node.
```

Comment:
Please move this into a utility method where we can improve it later. It's pretty nasty to need to do an O(np) search to find a workspace.